File: FindABook/Models/Book.cs
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace FindABook.Models
{
    public class Book
    {
        [Key]
        public int BookId { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public double Price { get; set; }

        public int CategoryId { get; set; }

        public bool BestSeller { get; set; }

        public string Publisher { get; set; }

        public string ImageUrl { get; set; }

        
        public virtual Category Category { get; set; }

        public virtual ICollection<BookSubscription> BookSubscription { get; set; }
    }
}

File: FindABook/ClientApp/src/app/interfaces/book.ts
import { Category } from "./category";

export class Book
{
    constructor(
        public Title :string,
        public Description :string,
        public Price :number,
        public CategoryId :number,
        public BestSeller :boolean,
        public Publisher : string,
        public ImageUrl: string, 
        public BookId?:number,   
        public Category?: Category,
        public BookSubscription? :any []
    ){

    }
    
}
